### PR TITLE
Add release workflows

### PR DESCRIPTION
**Pull request by jtherrmann**
TODO:

- [x] How to add all the required labels (major, minor, patch, bumpless)?
- [x] Do we need a deploy workflow to call `reusable-bump-version.yml`?
- [x] Do we want static analysis? (I don't think so?)

### CODE REVIEW

**Comment by jhkennedy** [comment]
@jtherrmann for your todos/questions:
>  How to add all the required labels (major, minor, patch, bumpless)?

For *new* repositories, they will come with required labels as the default labels are set at the org-level here: 
https://github.com/organizations/ASFHyP3/settings/repository-defaults

For repos that already exist and don't have them, you'll need to add them manually -- easiest to copy paste values from the org defaults. For this repo, you add them here:
https://github.com/organizations/ASFHyP3/settings/repository-defaults

>  Do we need a deploy workflow to call reusable-bump-version.yml?
Yes, we need a [`tag-version.yml`](https://github.com/ASFHyP3/hyp3-sdk/blob/develop/.github/workflows/tag-version.yml). The release workflow you've added is triggered on a new tag. To create the new tag from a merge to main, you need the reusable-bumb-version workflow. 

We technically could *not* use a `main` branch and just manually push a tag for a release, but then we wouldn't need the labeled PR workflow and there would be no release review mechanism.

> Do we want static analysis? (I don't think so?)

It's probably a good idea to have the secrets scanning on, but I don't think there's any linting or anything like that that's needed.

**Comment by jhkennedy** [comment]
I went ahead and added the labels.

File: .github/workflows/labeled-pr.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+      - opened
+      - labeled
+      - unlabeled
+      - synchronize
+    branches:
+      - main
+
+jobs:
+  call-labeled-pr-check-workflow:
+    uses: ASFHyP3/actions/.github/workflows/reusable-labeled-pr-check.yml@v0.7.0
```
Comment (jhkennedy) [inline .github/workflows/labeled-pr.yml:15]: I think the latest version in v0.7.1 but we could let dependabot bump all of these. Speaking of which, we'll want the dependabot yaml to keep the actions up to date:
https://github.com/ASFHyP3/hyp3-sdk/blob/develop/.github/dependabot.yml

**Comment by jtherrmann** [comment]
I've reviewed the trufflehog failures and confirmed that they can be safely ignored.

**Review by jhkennedy — APPROVED** [review]

**Pull request merged**